### PR TITLE
Butterfly and bumblebee census result services improvements

### PR DESCRIPTION
**Pull request by MLumme**
Branch at this point contains some bugfixes for the result services which would be good to merge now, but some tasks are not done, but they can be merged in a different pull request later. Branch build here: https://179339504.dev.laji.fi/, pivotal task with bug descriptions here: https://www.pivotaltracker.com/story/show/179339504

### CODE REVIEW

File: projects/laji/src/app/+project-form/results/syke-insect-result/syke-insect-route-table/syke-insect-route-table.component.ts [hunk 1]
```diff
@@ -13,10 +13,11 @@ import { DateFormatPipe } from 'ngx-moment';
   styleUrls: ['./syke-insect-route-table.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class SykeInsectRouteTableComponent {
+export class SykeInsectRouteTableComponent implements OnInit, OnChanges {
   @Input() routeId: string;
   @Input() season: string;
   @Input() sorts: {prop: string, dir: 'asc'|'desc'}[] = [];
+  @Input() activeYear;
```
Comment (aorin) [inline projects/laji/src/app/+project-form/results/syke-insect-result/syke-insect-route-table/syke-insect-route-table.component.ts:20]: This is missing a type. The taxonSet variable below seems to also be missing it

**Review by aorin — CHANGES_REQUESTED** [review]
Looks good, one small comment

File: projects/laji/src/app/+project-form/results/syke-insect-result/syke-insect-route-table/syke-insect-route-table.component.ts (hunk 1 above)
Comment (MLumme) [reply projects/laji/src/app/+project-form/results/syke-insect-result/syke-insect-route-table/syke-insect-route-table.component.ts]: Fixed by removing both, they were remnants from testing and should not have been left there.

Comment (aorin) [reply projects/laji/src/app/+project-form/results/syke-insect-result/syke-insect-route-table/syke-insect-route-table.component.ts]: If I'm looking at correct version, taxonSet is still used in the `syke-insect-route-table.component.html` template (2nd line)

Comment (MLumme) [reply projects/laji/src/app/+project-form/results/syke-insect-result/syke-insect-route-table/syke-insect-route-table.component.ts]: Right you are, I missed that use, returning it and setting it's type which should be the right one.

Comment (aorin) [reply projects/laji/src/app/+project-form/results/syke-insect-result/syke-insect-route-table/syke-insect-route-table.component.ts]: :+1:

**Review by aorin — APPROVED** [review]

**Pull request merged**